lib/msal-common/src/authority/Authority.ts

```diff
@@ -51,14 +51,17 @@ export class Authority {
     private proxyUrl: string;
     // Logger object
     private logger: Logger;
+    // Custom Http Agent
+    private customAgentOptions: CustomAgentOptions;
 
     constructor(
         authority: string,
         networkInterface: INetworkModule,
         cacheManager: ICacheManager,
         authorityOptions: AuthorityOptions,
         logger: Logger,
-        proxyUrl?: string
+        proxyUrl?: string,
```

Comment:
Seems a bit odd that Authority class has to change every time we add some field related to HTTP. This violates the S and O principles of SOLID (https://en.wikipedia.org/wiki/SOLID).

I would expect that all these details, including proxyUrl and agent options to be encapsulated inside `networkInterface` (without changing  INetworkModule interface, as that's a breaking change).

Reply:
I'm not sure I understand the second part of your comment. The network interface is the (optionally) user-custom-defined HttpClient. The Authority class has changed because we need to pass the proxyUrl/customAgentOptions from the (optional) user defined system config to the HttpClient.

Reply:
Tend to agree here - these sorts of details should be handled by the network interface rather than the caller. Ideally the caller (Authority in this case) shouldn't be exposed to this or expected to understand what to do with this @Robbie-Microsoft The network interface *can* be a custom defined interface, however, if it is not provided a default (our implementation of HttpClient) is passed in instead.